src/timer.rs

```diff
@@ -1,13 +1,12 @@
 //! Timers
 
-use core::convert::Infallible;
 use nb;
 
 /// A count down timer
 ///
 /// # Contract
 ///
-/// - `self.start(count); block!(self.wait());` MUST block for AT LEAST the time specified by
```

Comment:
Like with the busy-wait functions, can a timer ever actually fail? Should this be infallible? I'm less convinced than with the busy-waits, but I could still go either way on this.

Reply:
For completeness I repeat here: I also wondered about these traits. However, blocking behavior is not the same as a busy-wait implementation.